src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs

```diff
@@ -45,41 +47,61 @@ public static SwaggerFormattedReference FormatReferenceFullPath(string reference
             // Decode for URI Fragment Identifier Representation
             if (reference.StartsWith("#/"))
             {
-                // Reuse relative path, to decode the values inside '/'.
-                var path = reference.Substring(2);
-                var decodedPath = ((RelativePath)path).UrlDecode();
+                var result = ParseReferencePath(reference.Substring(2));
                 return new SwaggerFormattedReference
                 {
                     Type = SwaggerFormattedReferenceType.InternalReference,
-                    Path = "/" + decodedPath,
-                    Name = decodedPath.FileName
+                    Path = "/" + result.Item1,
+                    Name = result.Item2
                 };
             }
 
-            // External reference json
-            if (reference.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+            // Not decode for JSON String Representation
+            if (reference.StartsWith("/"))
             {
                 return new SwaggerFormattedReference
                 {
-                    Type = SwaggerFormattedReferenceType.ExternalReference,
+                    Type = SwaggerFormattedReferenceType.InternalReference,
                     Path = reference,
-                    Name = Path.GetFileNameWithoutExtension(reference)
+                    Name = reference.Split('/').Last()
                 };
             }
 
-            // Not decode for JSON String Representation
-            if (reference.StartsWith("/"))
+            // External reference
+            if (PathUtility.IsRelativePath(reference) && reference.IndexOf(JsonExtension, StringComparison.OrdinalIgnoreCase) >= 0)
             {
-                var fileName = reference.Split('/').Last();
-                return new SwaggerFormattedReference
+                // For example "file.json"
+                if (reference.EndsWith(JsonExtension, StringComparison.OrdinalIgnoreCase))
                 {
-                    Type = SwaggerFormattedReferenceType.InternalReference,
-                    Path = reference,
-                    Name = fileName
-                };
+                    return new SwaggerFormattedReference
+                    {
+                        Type = SwaggerFormattedReferenceType.ExternalReference,
+                        ExternalFilePath = reference,
+                        Name = Path.GetFileNameWithoutExtension(reference)
+                    };
+                }
+
+                // For example "file.json#/definitions/reference"
+                if (reference.Contains("#"))
+                {
+                    var values = reference.Split('#');
+                    if (values.Length != 2)
+                    {
+                        throw new InvalidOperationException($"Reference path '{reference}' should contain only one '#' character.");
+                    }
+                    var externalFilePath = ParseReferencePath(values[0]).Item1;
```

Comment:
yes better check, I'll add.

---
In reply to: [101673560](https://github.com/dotnet/docfx/pull/1308#discussion_r101673560) [](ancestors = 101673560)